Orange/widgets/data/owselectrows.py

```diff
@@ -809,5 +880,73 @@ def resizeEvent(self, QResizeEvent):
         self.set_text()
 
 
+class DateTimeWidget(QDateTimeEdit):
+    def __init__(self, parent, col_idx, datetime_format):
+        QDateTimeEdit.__init__(self, parent)
+
+        self.parent = parent
+        self.format = datetime_format
+        self.have_date, self.have_time = datetime_format[0], datetime_format[1]
+        self.column = parent.data[:, col_idx]
+        self.set_format()
+        self.setSizePolicy(
+            QSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding))
+        self.dateTimeChanged.connect(parent._invalidate_dates)
```

Comment:
Not like this. The idea of signals is that the object that needs to be informed about something "subscribes" to the signal. Remove this line and instead connect to the signal when you construct the widget, e.g.

```python
                w = DateTimeWidget(self, var_idx, datetime_format)
                w.dateTimeChanged.connect(self._invalidate_dates)
```

As it is now, this widget assumes that the parent has a method `_invalidate_dates` (and, besides, it starts with underscore, so it's kind of private). With the signals, method naming is up to parent. And nobody touches its private methods.